git_machete/cli.py

```diff
@@ -0,0 +1,540 @@
+#!/usr/bin/env python
+
+import getopt
+import os
+import re
+import sys
+import textwrap
+from typing import Dict, List, Optional, Tuple, TypeVar
+
+from git_machete import __version__
+from git_machete import utils
+from git_machete.client import MacheteClient, alias_by_command, allowed_directions
+from git_machete.constants import EscapeCodes
+from git_machete.docs import short_docs, long_docs
+from git_machete.exceptions import MacheteException, StopTraversal
+from git_machete.git_operations import GitContext
+from git_machete.options import CommandLineOptions
+from git_machete.utils import fmt, underline, excluding, warn
+
+T = TypeVar('T')
+
+initial_current_directory: Optional[str] = utils.get_current_directory_or_none() or os.getenv('PWD')
+
+command_by_alias: Dict[str, str] = {v: k for k, v in alias_by_command.items()}
+
+command_groups: List[Tuple[str, List[str]]] = [
+    # 'is-managed' is mostly for scripting use and therefore skipped
+    ("General topics", ["file", "help", "hooks", "version"]),
+    ("Build, display and modify the tree of branch dependencies",
+     ["add", "anno", "discover", "edit", "status"]),
+    ("List, check out and delete branches", ["delete-unmanaged", "go", "list", "show"]),
+    ("Determine changes specific to the given branch", ["diff", "fork-point", "log"]),
+    ("Update git history in accordance with the tree of branch dependencies",
+     ["advance", "reapply", "slide-out", "squash", "traverse", "update"])
+]
+
+
+def usage(command: str = None) -> None:
+    if command and command in command_by_alias:
+        command = command_by_alias[command]
+    if command and command in long_docs:
+        print(fmt(textwrap.dedent(long_docs[command])))
+    else:
+        print()
+        short_usage()
+        if command:
```

Comment:
This is a bit modified in comparison to the content of its ancestor `cmd.py`.
Previously it was if `command and command not in long_docs`. I changed it, because in `else` itself we've already verified that it's either `command == None/''/False` or `command exists and  it is not in long_docs`. Based on this I think it's redundant to check that `command exists and is not in long doc`s. It should be sufficient to just check if it `exists`, because if `it exists and is in the long_docs `it wouldn't enter the `else` part.